entities/storobj/storage_object.go

```diff
@@ -370,96 +371,130 @@ func (ko *Object) MarshalBinary() ([]byte, error) {
 	}
 	vectorWeightsLength := uint32(len(vectorWeights))
 
-	ec := &errorcompounder.ErrorCompounder{}
-	buf := bytes.NewBuffer(nil)
-	le := binary.LittleEndian
-	ec.Add(binary.Write(buf, le, &ko.MarshallerVersion))
-	ec.Add(binary.Write(buf, le, &ko.docID))
-	ec.Add(binary.Write(buf, le, kindByte))
-	_, err = buf.Write(idBytes)
-	ec.Add(err)
-	ec.Add(binary.Write(buf, le, ko.CreationTimeUnix()))
-	ec.Add(binary.Write(buf, le, ko.LastUpdateTimeUnix()))
-	ec.Add(binary.Write(buf, le, vectorLength))
-	ec.Add(binary.Write(buf, le, ko.Vector))
-	ec.Add(binary.Write(buf, le, classNameLength))
-	_, err = buf.Write(className)
-	ec.Add(err)
-	ec.Add(binary.Write(buf, le, schemaLength))
-	_, err = buf.Write(schema)
-	ec.Add(err)
-	ec.Add(binary.Write(buf, le, metaLength))
-	_, err = buf.Write(meta)
-	ec.Add(err)
-	ec.Add(binary.Write(buf, le, vectorWeightsLength))
-	_, err = buf.Write(vectorWeights)
-	ec.Add(err)
-
-	return buf.Bytes(), ec.ToError()
+	totalBufferLength := 1 + 8 + 1 + 16 + 8 + 8 + 2 + vectorLength*4 + 2 + classNameLength + 4 + schemaLength + 4 + metaLength + 4 + vectorWeightsLength
+	byteBuffer := make([]byte, totalBufferLength)
+	byteBuffer[0] = ko.MarshallerVersion
```

Comment:
maybe `WriteByte` method can be created for such cases, which also moves Position by 1

Reply:
Nice idea, added